src/widgets/barchart/bar.rs

```diff
@@ -56,6 +56,40 @@ impl<'a> Bar<'a> {
         self
     }
 
+    pub(super) fn render_value_with_different_styles(
+        self,
+        buf: &mut Buffer,
+        area: Rect,
+        bar_size: usize,
+        default_value_style: Style,
+        bar_style: Style,
+    ) {
+        let text = if let Some(text) = self.text_value {
+            text
+        } else {
+            self.value.to_string()
+        };
+
+        if !text.is_empty() {
+            let style = self.value_style.patch(default_value_style);
```

Comment:
I don't understand this line - the parameter that's passed to the method is `self.value_style`, so doesn't this just mean `self.value_style.patch(self.value_style)`?

A unit test for this method would help clarify the expected rendered output.

Reply:
one `self` has the type `Bar` and he other one has the type `BarChart`. So the styles are different.

but since you mentioned it somewhere else. this line should be:

`let style = default_value_style.patch(self.value_style);`

I will create unit tests for all styles, since patching the style is everywhere wrong / (swapped). :smile: